src/Interpreters/Context.h

```diff
@@ -201,6 +201,20 @@ class Context
 
     QueryAccessInfo query_access_info;
 
+    struct QueryFactoriesInfo
+    {
+        std::set<std::string> aggregate_functions;
+        std::set<std::string> databases;
+        std::set<std::string> data_types;
+        std::set<std::string> dictionaries;
+        std::set<std::string> formats;
+        std::set<std::string> functions;
```

Comment:
There is a question - should we add canonical function name (e.g. substring) or exactly what user specified (e.g. `SUBSTRING`). I think that for the purpose of test coverage we should collect canonical names.

The same for AggregateFunctionFactory. For example, if user specified `SUMIf`, I think we should record the usage of `sum` and the usage of AggregateFunctionCombinator `If`.